mux.go

```diff
@@ -45,9 +45,14 @@ func (s *Server) checkAuth(r *http.Request) (username string, ok bool) {
 	return username, true
 }
 
-func (s *Server) wrapAuth(f http.HandlerFunc) http.HandlerFunc {
+func (s *Server) wrapMetricsAuth(f http.HandlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		if _, ok := s.checkAuth(r); !ok {
+		username, ok := s.checkAuth(r)
+		if !ok {
+			httpDefaultError(w, http.StatusUnauthorized)
+			return
+		}
+		if s.PrivateRepos && username != "metrics" {
```

Comment:
This will need some kind of documentation, otherwise users will be very confused

Reply:
This is actually to stay compatible with previous rest-server releases:

- If PrivateRepos is false, accept any valid auth to access metrics.
- If PrivateRepos is true, only user 'metrics' can access this endpoint, which is a side-effect of how we currently do auth.
- There is a new PrometheusNoAuth (`--prometheus-no-auth`) option to disable auth on this endpoint. 

I plan to add a config option to serve the metrics on a different address instead, which I should have done in the first place.

Reply:
Understood, thanks!